lib/actions/api.js

```diff
@@ -83,86 +82,96 @@ function getActiveItinerary (otpState) {
-    // involving realtime info embedded in the OTP response.
-    // (That action records an entry again in the middleware.)
-    // For users who opted in to store trip request history,
-    // to avoid recording the same trip request twice in the middleware,
-    // only add the user Authorization token to the request
-    // when querying the non-realtime route.
-    //
-    // The advantage of using non-realtime route is that the middleware will be able to
-    // record and provide the theoretical itinerary summary without having to query OTP again.
-    // FIXME: Interestingly, and this could be from a side effect elsewhere,
-    // when a logged-in user refreshes the page, the trip request in the URL is not recorded again
-    // (state.user stays unpopulated until after this function is called).
-    //
-    const { user } = state
-    const storeTripHistory = user &&
-      user.loggedInUser &&
-      user.loggedInUser.storeTripHistory
-
-    fetch(constructRoutingQuery(otpState, true), getOtpFetchOptions(state, storeTripHistory))
-      .then(getJsonAndCheckResponse)
-      .then(json => {
-        // FIXME: This is only performed when ignoring realtimeupdates currently, just
-        // to ensure it is not repeated twice.
-        dispatch(nonRealtimeRoutingResponse({ response: json, searchId }))
-      })
-      .catch(error => {
-        console.error(error)
-        // do nothing
-      })
+      // Also fetch a non-realtime route.
+      //
+      // FIXME: The statement below may no longer apply with future work
+      // involving realtime info embedded in the OTP response.
+      // (That action records an entry again in the middleware.)
+      // For users who opted in to store trip request history,
+      // to avoid recording the same trip request twice in the middleware,
+      // only add the user Authorization token to the request
+      // when querying the non-realtime route.
+      //
+      // The advantage of using non-realtime route is that the middleware will be able to
+      // record and provide the theoretical itinerary summary without having to query OTP again.
+      // FIXME: Interestingly, and this could be from a side effect elsewhere,
+      // when a logged-in user refreshes the page, the trip request in the URL is not recorded again
+      // (state.user stays unpopulated until after this function is called).
+      //
+      const { user } = state
+      const storeTripHistory = user &&
+        user.loggedInUser &&
+        user.loggedInUser.storeTripHistory
+
+      fetch(constructRoutingQuery(otpState, true), getOtpFetchOptions(state, storeTripHistory))
+        .then(getJsonAndCheckResponse)
+        .then(json => {
+          // FIXME: This is only performed when ignoring realtimeupdates currently, just
+          // to ensure it is not repeated twice.
+          dispatch(nonRealtimeRoutingResponse({ response: json, searchId }))
+        })
```

Comment:
Since we're now making queries for all kinds of mode combinations, we should probably check some of the mode combinations that won't have realtime data and avoid making requests for those types of queries.

Reply:
Will add FIXME for now.